### PR TITLE
Make data connection more stable and timeout available

### PR DESCRIPTION
**Pull request by heat1024**
Modify data_handler's logic for stable data transfer.

Changes

- handle data connection execution FTP commands by proxy
- use mutex before use socket conn variable
- some flags change to atomic variable from boolean for resolve data races
- data transfer now use manual packet copy instead io.CopyBuffer for increase timeout by data transfer
- fix login flag set place. so user can login without disconnect if failed first login by wrong password

### CODE REVIEW

File: pftp/handle_commands.go [hunk 1]
```diff
@@ -161,8 +163,8 @@ func (c *clientHandler) handlePBSZ() *result {
 
 // response PROT to client and store command line when connect by TLS & not loggined
 func (c *clientHandler) handlePROT() *result {
-	if c.controlInTLS {
-		if !c.isLoggedin {
+	if atomic.LoadInt32(&c.controlInTLS) == 1 {
```
Comment (pyama86) [inline pftp/handle_commands.go:166]: you had better to make control structs, like a https://gist.github.com/mahan/6256149

Comment (heat1024) [reply pftp/handle_commands.go:166]: I looks easier for code read!

Comment (heat1024) [reply pftp/handle_commands.go]: https://github.com/pyama86/pftp/pull/45/commits/0782994610d7ae275b46b10769361c23a6eabb30

replace atomic operation to abool module.

**Review by pyama86 — APPROVED** [review]
thanks!

**Pull request merged**